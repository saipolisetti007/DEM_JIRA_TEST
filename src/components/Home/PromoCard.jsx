import React from 'react';
import { Box } from '@mui/material';

const PromoCard = ({ backgroundImage, backgroundColor, children }) => {
  return (
    <Box
      sx={{
        position: 'relative',
        width: '49%',
        height: '100%',
        minHeight: '250px',
        backgroundImage: `url(${backgroundImage})`,
        backgroundRepeat: 'no-repeat',
        backgroundSize: 'cover',
        backgroundPosition: 'center',
        color: backgroundColor,
        borderRadius: '10px',
        '&::after': {
          content: '""',
          position: 'absolute',
          top: 0,
          left: 0,
          width: '100%',
          height: '100%',
          backgroundColor: backgroundColor,
          opacity: 0.9,
          zIndex: 1,
          borderRadius: '10px',
          mixBlendMode: 'multiply'
        },
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        zIndex: 0
      }}>
      <Box
        sx={{
          display: 'grid',
          gap: 3,
          width: '100%',
          position: 'relative',
          zIndex: 2,
          color: 'white',
          borderRadius: '10px'
        }}>
        {children}
      </Box>
    </Box>
  );
};

export default PromoCard;
