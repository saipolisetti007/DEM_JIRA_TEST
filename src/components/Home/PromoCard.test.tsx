import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import PromoCard from './PromoCard';

describe('PromoCard', () => {
  test('should render the component with children correctly', () => {
    render(
      <PromoCard backgroundImage="test-image.jpg" backgroundColor="#000">
        <div>Test Child</div>
      </PromoCard>
    );

    expect(screen.getByText('Test Child')).toBeInTheDocument();
  });

  test('should apply the correct background image and color styles', () => {
    const { container } = render(
      <PromoCard backgroundImage="test-image.jpg" backgroundColor="#000">
        <div>Test Child</div>
      </PromoCard>
    );

    const box = container.firstChild;
    expect(box).toHaveStyle(`
      background-image: url(test-image.jpg);
      color: #000;
    `);
    // Check if the child element is rendered correctly
    if (box) {
      expect(box.firstChild).toHaveTextContent('Test Child');
    }
  });
});
