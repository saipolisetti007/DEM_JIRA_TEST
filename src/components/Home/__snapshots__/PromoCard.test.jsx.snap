// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromoCard should apply the correct background image and color styles 1`] = `
<div
  class="MuiBox-root css-1ouuv4p"
>
  <div
    class="MuiBox-root css-1mjmca5"
  >
    <div>
      Test Child
    </div>
  </div>
</div>
`;
